instant-xml/src/impls.rs

```diff
@@ -769,3 +770,23 @@ impl<'xml> FromXml<'xml> for IpAddr {
 
     const KIND: Kind = Kind::Scalar;
 }
+
+#[cfg(test)]
+mod tests {
+    use super::decode;
+
+    #[test]
+    fn test_decode() {
+        assert_eq!(decode("foo").unwrap(), "foo");
+        assert_eq!(decode("foo &amp; bar").unwrap(), "foo & bar");
+        assert_eq!(decode("foo &lt; bar").unwrap(), "foo < bar");
+        assert_eq!(decode("foo &gt; bar").unwrap(), "foo > bar");
+        assert_eq!(decode("foo &quot; bar").unwrap(), "foo \" bar");
+        assert_eq!(decode("foo &apos; bar").unwrap(), "foo ' bar");
+        assert_eq!(decode("foo &amp;lt; bar").unwrap(), "foo &lt; bar");
+        assert_eq!(decode("&amp; foo").unwrap(), "& foo");
+        assert_eq!(decode("foo &amp;").unwrap(), "foo &");
+        assert!(decode("&foo;").is_err());
+        assert!(decode("&foobar;").is_err());
+    }
```

Comment:
Let's add some more utf8 values here to be safe.
* `cbdtéda&amp;sü` -> `cbdtéda&sü`
* `cbdtéd&ampü` -> should error I guess

Reply:
Cool, one of those actually found a bug. Changed `decode()` to work in the byte domain, which simplifies the non-error path code and seems to be slightly faster too.